app/templates/main/map.html

```diff
@@ -106,509 +111,630 @@
+          } else {
+            obj = processData(uploadURL, boxes, $('#email').val(), $('#startDay').val(), $('#endDay').val());
+            inside = new Array(obj.boxes.length).fill(false);
+            startTimes = new Array(obj.boxes.length).fill(null);
+            durations = new Array(obj.boxes.length).fill(0);
+            var newOboe = oboe();
+
+            newOboe.node('locations.*', function (location) {
+              if (weekEndMS === 0) {
+                weekEndMS = moment.unix(location.timestampMs / 1000).day(obj.dayEnd).hour(23).minute(59).unix() * 1000;
+                weekStartMS = moment.unix(location.timestampMs / 1000).day(obj.dayStart).hour(0).minute(0).unix() * 1000;
+              }
+
+              if (location.timestampMs >= weekStartMS && location.timestampMs <= weekEndMS) {
+                // within a week
+                for (var i = 0; i < obj.boxes.length; i++) {
+                  // iterate thru the boxes
+                  var box = obj.boxes[i];
+                  var lat = location.latitudeE7 / Math.pow(10, 7)
+                  var lng = location.longitudeE7 / Math.pow(10, 7)
+                    // check if in box coors && not inside a box
+                  if (box['swLat'] <= lat && box['neLat'] >= lat &&
+                    box['swLng'] <= lng && box['neLng'] >= lng && !inside[i]) {
+                    console.log('enetered ' + box.name);
+                    inside[i] = true;
+                    startTimes[i] = location.timestampMs;
+                  } else if ((box['swLat'] <= lat && box['neLat'] >= lat &&
+                      box['swLng'] <= lng && box['neLng'] >= lng) && !inside[i]) {
+                    durations[i] = Math.abs(location.timestampMs - startTimes[i]);
+                  }
+                  // not inside the box
+                  else if (!(box['swLat'] <= lat && box['neLat'] >= lat &&
+                      box['swLng'] <= lng && box['neLng'] >= lng) && inside[i]) {
+                    inside[i] = false;
+                    durations[i] = Math.abs(location.timestampMs - startTimes[i]);
+                  }
+                }
+
+              }
+              else if (location.timestampMs < weekStartMS) {
+                console.log(moment.unix(location.timestampMs / 1000).format('MM-DD-YYYY ddd HH:mm'));
+                weekEndMS -= 604800000
+                weekStartMS -= 604800000
+                var locationtotal = 0;
+                for (var i = 0; i < obj.boxes.length; i++) {
+                  if (startTimes[i] && durations[i] > 0) {
+                    console.log('pushed!' + startTimes[i]);
+                    csv.push([obj.boxes[i].name, moment.unix(startTimes[i] / 1000).format('MM-DD-YYYY ddd HH:mm'),
+                      (durations[i] / 1000 / 3600).toFixed(2) + " hrs"
+                    ])
+                    startTimes[i] = weekEndMS;
```

Comment:
why does this need to be updated here? is this in the case the worker stays in the bounding box while a week ends and the new work week begins?